server/controllers/users.js

```diff
@@ -40,9 +40,9 @@ exports.getAll = async (req, res) => {
     }
 }
 
-exports.getById = async (req, res) => {
+exports.getByUsername = async (req, res) => {
     try {
-        const result = await User.findById(req.params.id)
+        const result = await User.findOne({ username: req.body.username })
```

Comment:
get username from reqeust params instead of the body
change routes file accordingly to the changes made